pkg/systemlogmonitor/log_monitor.go

```diff
@@ -73,13 +73,35 @@ func NewLogMonitorOrDie(configPath string) types.Monitor {
 		glog.Fatalf("Failed to validate matching rules %+v: %v", l.config.Rules, err)
 	}
 	glog.Infof("Finish parsing log monitor config file: %+v", l.config)
+
 	l.watcher = logwatchers.GetLogWatcherOrDie(l.config.WatcherConfig)
 	l.buffer = NewLogBuffer(l.config.BufferSize)
 	// A 1000 size channel should be big enough.
 	l.output = make(chan *types.Status, 1000)
+
+	initializeProblemMetricsOrDie(l.config.Rules)
```

Comment:
Should we call this in the monitor loop, alongside with `l.initializeStatus()`?

Reply:
I prefer not. Mainly because I prefer any function that ends in `OrDie` (i.e. may panic when en-counting error) happens before `Start()`.

And if declaring/initializing a metric failed, I actually prefer NPD to just panic. Because that's most likely a configuration problem. And I prefer the failure to be very loud.

Reply:
ACK